nf_core/subworkflows/lint/subworkflow_changes.py

```diff
@@ -20,7 +23,29 @@ def subworkflow_changes(subworkflow_lint_object, subworkflow):
 
     Only runs when linting a pipeline, not the modules repository
     """
-    tempdir = subworkflow.component_dir
+    if subworkflow.is_patched:
+        # If the subworkflow is patched, we need to apply
+        # the patch in reverse before comparing with the remote
+        tempdir_parent = Path(tempfile.mkdtemp())
+        tempdir = tempdir_parent / "tmp_subworkflow_dir"
+        shutil.copytree(subworkflow.component_dir, tempdir)
+        try:
+            new_lines = ComponentsDiffer.try_apply_patch(
+                subworkflow.component_type,
+                subworkflow.component_name,
+                subworkflow.org,
+                subworkflow.patch_path,
+                tempdir,
+                reverse=True,
+            )
+            for file, lines in new_lines.items():
+                with open(tempdir / file, "w") as fh:
+                    fh.writelines(lines)
+        except LookupError:
+            # This error is already reported by subworkflow_patch, so just return
```

Comment:
wtihout checking, this feels like we would also use the same logic inn `modules_changes`. can we make a `components_changes` superfunction?